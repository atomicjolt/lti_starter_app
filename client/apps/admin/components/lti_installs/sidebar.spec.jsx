import React from 'react';
import TestRenderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Sidebar from './sidebar';

const mockStore = configureStore([]);
const store = mockStore({});

describe('lti installs sidebar', () => {
  let result;
  let instance;
  const applicationName = 'applicationName';
  const accounts = {
    1: {
      id: 1,
      name: 'accountName',
      sub_accounts: []
    }
  };
  const application = {
    name: applicationName
  };
  const applicationInstance = {
    site: {
      url: 'www.atomicjolt.com'
    }
  };
  const saveApplicationInstance = () => {};
  const canvasRequest = () => {};
  const setAccountActive = () => {};
  const sites = {};

  describe('should render sidebar', () => {
    beforeEach(() => {
      result = TestRenderer.create(
        <Provider store={store}>
          <Sidebar
            accounts={accounts}
            application={application}
            applicationInstance={applicationInstance}
            saveApplicationInstance={saveApplicationInstance}
            canvasRequest={canvasRequest}
            setAccountActive={setAccountActive}
            sites={sites}
          />
        </Provider>
      );
      instance = result.root;
    });

    it('renders', () => {
      expect(result).toBeDefined();
    });

    it('renders the form not null', () => {
      expect(result).not.toBeNull();
    });

    it('matches the snapshot', () => {
      expect(result).toMatchSnapshot();
    });

    it('return the title name', () => {
      const title = instance.findByProps({ className: 'c-tool__title' });
      expect(title.props.children).toBe(applicationName);
    });

    it('return the site url', () => {
      const title = instance.findByProps({ className: 'c-tool__instance' });
      expect(title.props.children.props.href).toEqual('www.atomicjolt.com');
    });

  });
});
