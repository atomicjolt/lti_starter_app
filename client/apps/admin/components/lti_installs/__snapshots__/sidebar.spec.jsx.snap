// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lti installs sidebar should render sidebar matches the snapshot 1`] = `
<div
  className="o-left"
>
  <div
    className="c-tool"
  >
    <span>
      <a
        onClick={[Function]}
      >
        <i
          className="material-icons"
        >
          settings
        </i>
      </a>
    </span>
    <h4
      className="c-tool__subtitle"
    >
      LTI Tool
    </h4>
    <h3
      className="c-tool__title"
    >
      applicationName
    </h3>
  </div>
  <div
    className="c-tool"
  >
    <h4
      className="c-tool__instance"
    >
      <a
        href="www.atomicjolt.com"
      >
        www.atomicjolt.com
      </a>
    </h4>
  </div>
  <div
    className="c-filters"
  >
    <h4
      className="c-sidebar-subtitle"
    >
      Accounts
    </h4>
    <Accounts
      accounts={
        Object {
          "1": Object {
            "id": 1,
            "name": "accountName",
            "sub_accounts": Array [],
          },
        }
      }
      setAccountActive={[Function]}
    />
  </div>
</div>
`;
